src/libraries/System.Net.Security/src/System/Net/Security/NetSecurityTelemetry.cs

```diff
@@ -0,0 +1,288 @@
+                {
+                    DisplayName = "Total TLS handshakes completed"
+                };
+
+                _currentTlsHandshakesCounter ??= new PollingCounter("current-tls-handshakes", this, () => -Interlocked.Read(ref _finishedTlsHandshakes) + Interlocked.Read(ref _startedTlsHandshakes))
+                {
+                    DisplayName = "Current TLS handshakes"
+                };
+
+                _failedTlsHandshakesCounter ??= new PollingCounter("failed-tls-handshakes", this, () => Interlocked.Read(ref _failedTlsHandshakes))
+                {
+                    DisplayName = "Total TLS handshakes failed"
+                };
+
+                _connectionsOpenTls10Counter ??= new PollingCounter("tls10-connections-open", this, () => Interlocked.Read(ref _connectionsOpenTls10))
+                {
+                    DisplayName = "TLS 1.0 Connections Active"
+                };
+
+                _connectionsOpenTls11Counter ??= new PollingCounter("tls11-connections-open", this, () => Interlocked.Read(ref _connectionsOpenTls11))
+                {
+                    DisplayName = "TLS 1.1 Connections Active"
+                };
+
+                _connectionsOpenTls12Counter ??= new PollingCounter("tls12-connections-open", this, () => Interlocked.Read(ref _connectionsOpenTls12))
+                {
+                    DisplayName = "TLS 1.2 Connections Active"
+                };
+
+                _connectionsOpenTls13Counter ??= new PollingCounter("tls13-connections-open", this, () => Interlocked.Read(ref _connectionsOpenTls13))
+                {
+                    DisplayName = "TLS 1.3 Connections Active"
+                };
+
+                _handshakeDurationTls10Counter ??= new EventCounter("tls10-handshake-duration", this)
+                {
+                    DisplayName = "TLS 1.0 Handshake Duration",
+                    DisplayUnits = "ms"
+                };
+
+                _handshakeDurationTls11Counter ??= new EventCounter("tls11-handshake-duration", this)
+                {
+                    DisplayName = "TLS 1.1 Handshake Duration",
+                    DisplayUnits = "ms"
+                };
+
+                _handshakeDurationTls12Counter ??= new EventCounter("tls12-handshake-duration", this)
+                {
+                    DisplayName = "TLS 1.2 Handshake Duration",
+                    DisplayUnits = "ms"
+                };
+
+                _handshakeDurationTls13Counter ??= new EventCounter("tls13-handshake-duration", this)
+                {
+                    DisplayName = "TLS 1.3 Handshake Duration",
+                    DisplayUnits = "ms"
+                };
+            }
+        }
+
+
+        [Event(1, Level = EventLevel.Informational)]
+        public void HandshakeStart(bool isServer, string targetHost)
+        {
+            Interlocked.Increment(ref _startedTlsHandshakes);
+
+            if (IsEnabled(EventLevel.Informational, EventKeywords.None))
```

Comment:
You can safely remove the `IsEnabled` checks inside these functions.  They will occur just inside of `WriteEvent`.  Arguments have already been created, so there isn't much perf to save here, especially if `WriteEvent` gets inlined.